src/main/java/org/ml_methods_group/algorithm/entity/RmmrEntitySearcher.java

```diff
@@ -0,0 +1,196 @@
+import com.intellij.psi.*;
+import org.apache.log4j.Logger;
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.ml_methods_group.algorithm.properties.finder_strategy.FinderStrategy;
+import org.ml_methods_group.algorithm.properties.finder_strategy.RmmrStrategy;
+import org.ml_methods_group.config.Logging;
+
+import java.util.*;
+
+public class RmmrEntitySearcher {
+    private static final Logger LOGGER = Logging.getLogger(EntitySearcher.class);
+
+    private final Map<String, PsiClass> classForName = new HashMap<>();
+    private final Map<PsiMethod, MethodEntity> entities = new HashMap<>();
+    private final Map<PsiClass, ClassEntity> classEntities = new HashMap<>();
+    private final AnalysisScope scope;
+    private final long startTime;
+    private final FinderStrategy strategy;
+    private final ProgressIndicator indicator;
+
+    private RmmrEntitySearcher(AnalysisScope scope) {
+        this.scope = scope;
+        strategy = RmmrStrategy.getInstance();
+        startTime = System.currentTimeMillis();
+        if (ProgressManager.getInstance().hasProgressIndicator()) {
+            indicator = ProgressManager.getInstance().getProgressIndicator();
+        } else {
+            indicator = new EmptyProgressIndicator();
+        }
+    }
+
+    @NotNull
+    public static EntitySearchResult analyze(AnalysisScope scope) {
+        final RmmrEntitySearcher finder = new RmmrEntitySearcher(scope);
+        return finder.runCalculations();
+    }
+
+    @NotNull
+    private EntitySearchResult runCalculations() {
+        indicator.pushState();
+        indicator.setText("Searching entities");
+        indicator.setIndeterminate(true);
+        LOGGER.info("Indexing entities...");
+        scope.accept(new UnitsFinder());
+        indicator.setIndeterminate(false);
+        LOGGER.info("Calculating properties...");
+        indicator.setText("Calculating properties");
+        scope.accept(new PropertiesCalculator());
+        indicator.popState();
+        return prepareResult();
+    }
+
+    @NotNull
+    private EntitySearchResult prepareResult() {
+        LOGGER.info("Preparing results...");
+        final List<ClassEntity> classes = new ArrayList<>();
+        final List<MethodEntity> methods = new ArrayList<>();
+        for (MethodEntity methodEntity : entities.values()) {
+            indicator.checkCanceled();
+            methods.add(methodEntity);
+        }
+        for (ClassEntity classEntity : classEntities.values()) {
+            indicator.checkCanceled();
+            classes.add(classEntity);
+        }
+        LOGGER.info("Properties calculated");
+        LOGGER.info("Generated " + classes.size() + " class entities");
+        LOGGER.info("Generated " + methods.size() + " method entities");
+        LOGGER.info("Generated " + 0 + " field entities. Fields are not supported.");
```

Comment:
does this line make any sense?

Reply:
I don't know. I report that this algorithm does not work with fields. I think it serves for consistency, because log information in EntitySearcher is structured the same.